### DISCUSSION TITLE
Can GeneralizedRCNN forward pass return both loss AND inference results?

### DISCUSSION THREAD

**Discussion by take5v** [facebookresearch/detectron2#4328] [Ideas]
Very often it is desirable to compute both loss function and different metrics using inference results on a validation set. For this, in a current version of `GeneralizedRCNN`, two methods should be called separately (`forward` and `inference`) requiring two network passes. Adding inference results to return values will increase validation time when both loss and metrics are computed.

**Comment by CloudyDory** [comment]
I would also like to see this functionality. However you can always look into the code and write a custom function that calculates both losses and inference results.
